scripts/jina-hub-update.py

```diff
@@ -2,97 +2,281 @@
-from github import Github
+from github import Github, PullRequestMergeStatus, Repository
+from github.Issue import Issue
+from github.PullRequest import PullRequest
 from ruamel.yaml import YAML
 
+WAIT_BETWEEN_PR_CHECKS = 5 * 60
+FIX_MODULE_TEMPLATE = 'fix module '
+TAG_IN_ISSUES = os.environ.get('TAG_IN_ISSUES', '')
+MODULES_REPO = os.environ.get('MODULES_REPO')
+if MODULES_REPO is None:
+    print(f'Error: MODULES_REPO needs to be set. Exiting...')
+    sys.exit(1)
+
 # this one has PR push access
 g = Github(os.environ["GITHUB_TOKEN"])
 
 yaml = YAML()
 
 
+def create_pr(fpath, module, jina_core_version, hub_repo, hub_origin, gh_hub_repo) -> Optional[PullRequest]:
+    """for each module with manifest.yml attempts to bump version and add jina core version.
+    then opens PR"""
+    pr = None
+    print(f'handling {module}...')
+    with open(fpath) as fp:
+        info = yaml.load(fp)
+        # make sure the (possibly) existing version is older
+        if 'jina_version' in info.keys():
+            # TODO should this be list?
+            existing_jina_version = info['jina_version']
+            if semver.VersionInfo.parse(existing_jina_version) >= semver.VersionInfo.parse(jina_core_version):
+                print(f'existing jina-core version for {module} was greater or equal than version to update '
+                      f'({existing_jina_version} >= {jina_core_version}). Skipping...')
+                return
+
+        old_ver = info['version']
+        new_ver = '.'.join(old_ver.split('.')[:-1] + [str(int(old_ver.split('.')[-1]) + 1)])
+        info['version'] = new_ver
+        print(f'bumped to {new_ver}')
+        info['jina_version'] = jina_core_version
+    with open(fpath, 'w') as fp:
+        yaml.dump(info, fp)
+
+    br_name = ''
+    try:
+        print('preparing the branch ...')
+        br_name = f'chore-{module.lower()}-{new_ver.replace(".", "-")}-core-{jina_core_version.replace(".", "-")}'
+        new_branch = hub_repo.create_head(br_name)
+        new_branch.checkout()
+
+        print(f'bumping version to {new_ver} and committing to {new_branch}...')
+        hub_repo.git.add(update=True)
+        # NOTE limited to 72 characters by commit lint
+        hub_repo.index.commit(f'chore: bump {module} version to {new_ver}')
+        hub_repo.git.push('--set-upstream', hub_origin, hub_repo.head.ref)
+
+        print('making a PR ...')
+        title_string = f'bumping version for {module} to {new_ver} (new jina: {jina_core_version})'
+        body_string = f'Due to the release of jina core v{jina_core_version}, this PR is automatically submitted to ' \
+                      f'ensure compatibility '
+
+        pr = gh_hub_repo.create_pull(
+            title=title_string,
+            body=body_string,
+            head=br_name,
+            base='master'
+        )
+    except git.GitCommandError as e:
+        print(f'Caught exception: {repr(e)}')
+        if 'tip of your current branch is behind' in str(e) \
```

Comment:
Why do we need to act just with these messages? can't we do the same for every exception?

Reply:
If the message ever change it will be quite problematic

Reply:
I refactored it. I now check if the PR exists before even trying to create the branch.